File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.cs
using System;

public class Solution 
{
    public int[] solution(string[] keyinput, int[] board) 
    {
        int[] playerPos = {0,0};
        
        int boundaryLeft  = -((board[0] - 1) / 2);
        int boundaryRight =  ((board[0] - 1) / 2);
        int boundaryUp    =  ((board[1] - 1) / 2);
        int boundaryDown  = -((board[1] - 1) / 2);
        for(int i=0; i<keyinput.Length; i++)
        {
            switch (keyinput[i])
            {
                case "left":
                    if(playerPos[0] == boundaryLeft) 
                    {
                        continue;
                    }
                    playerPos[0]--;
                    break;
                case "right":
                    if(playerPos[0] == boundaryRight) 
                    {
                        continue;
                    }
                    playerPos[0]++;
                    break;
                case "up":
                    if(playerPos[1] == boundaryUp) 
                    {
                        continue;
                    }
                    playerPos[1]++;
                    break;
                case "down":
                    if(playerPos[1] == boundaryDown) 
                    {
                        continue;
                    }
                    playerPos[1]--;
                    break;                   
            }
        }
        
        return playerPos;
    }
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> keyinput, vector<int> board) 
{
    int x = (board[0] - 1)/2;
    int y = (board[1] - 1)/2;
    vector<int> vec(2,0);
    for(string &ip : keyinput)
    {
        if(ip == "up")
        {
            if(vec[1] < y)
            {
                vec[1]++;
            }
        }
        else if(ip == "down")
        {
            if(vec[1] > -y)
            {
                vec[1]--;
            }
        }
        else if(ip == "left")
        {
            if(vec[0] > -x)
            {
                vec[0]--;
            }
        }
        else if(ip == "right")
        {
            if(vec[0] < x)
            {
                vec[0]++;
            }
        }
    }
    return vec;
}